backend/fms_core/services/workbook_utils.py

```diff
@@ -0,0 +1,32 @@
+from dataclasses import dataclass
+from typing import Callable, Iterable, Literal
+from openpyxl.cell.cell import Cell
+from openpyxl.worksheet.datavalidation import DataValidation
+from openpyxl.worksheet.worksheet import Worksheet
+
+@dataclass
+class CellDescription:
+    value: str
+    validation: DataValidation | None
+    apply_cell: Callable[[Cell], None] | None
+
+CD = CellDescription
+
+def insert_cells(worksheet: Worksheet, topleft: tuple[int, int], descriptors: Iterable[Iterable[CellDescription]], order: Literal["row"] | Literal["col"]):
```

Comment:
is topleft the best name you can find for this ?

Reply:
`initial_cell_location`?